walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +294,488 @@ public void notEnoughRoom() {
+      TimeRange.fromStartDuration(TIME_0930AM, DURATION_30_MINUTES),
+      TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true)
+      );
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noTimeOptionalAttendees() {
+    // Events  : |--A--|    |--A--|
+    // Events  :       |-B-|      |--B--|
+    // Day     : |----------------------|
+    // Options :           
+
+    Collection<Event> events = Arrays.asList(
+      new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+          Arrays.asList(PERSON_A)),
+      new Event("Event 2", TimeRange.fromStartDuration(TIME_1000AM, DURATION_1_HOUR),
+          Arrays.asList(PERSON_A)),
+      new Event("Event 3", TimeRange.fromStartDuration(TIME_0830AM, DURATION_90_MINUTES),
+          Arrays.asList(PERSON_B)),
+      new Event("Event 4", TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true),
+          Arrays.asList(PERSON_B)));
+    MeetingRequest request = new MeetingRequest(NO_ATTENDEES, DURATION_30_MINUTES);
+
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected = Arrays.asList(
+      TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+      TimeRange.fromStartDuration(TIME_1000AM, DURATION_1_HOUR),
+      TimeRange.fromStartDuration(TIME_0830AM, DURATION_90_MINUTES),
+      TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true)
+    );
+
+    Assert.assertTrue(expected.size() == actual.size() && 
+    expected.containsAll(actual) && actual.containsAll(expected));
+  }
+
+  @Test
+  public void onlyOneOptionalAttendeeMakesIt() {
+    // Events  : |--A--|    |--A--|
+    // Events  :       |-B-|      
+    // Events  : |----------C-----------|
+    // Day     : |----------------------|
+    // Options :                  |-----|
+
+    Collection<Event> events = Arrays.asList(
+      new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+          Arrays.asList(PERSON_A)),
+      new Event("Event 2", TimeRange.fromStartDuration(TIME_1000AM, DURATION_1_HOUR),
+          Arrays.asList(PERSON_A)),
+      new Event("Event 3", TimeRange.fromStartDuration(TIME_0830AM, DURATION_90_MINUTES),
+          Arrays.asList(PERSON_B)),
+      new Event("Event 4", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true),
+          Arrays.asList(PERSON_C)));
+    MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_30_MINUTES);
+
+    request.addOptionalAttendee(PERSON_B);
+    request.addOptionalAttendee(PERSON_C);
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected = Arrays.asList(TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void lotsOfAttendees() {
+
```

Comment:
Don't need the empty line here. The examples have empty lines for separating comments and code. Probably you should also add some comments here to describe the test case.